app/assets/javascripts/student_profile/QuadConverter.js

```diff
@@ -0,0 +1,103 @@
+import _ from 'lodash';
+import moment from 'moment';
+
+// A quad is a 4-element array of numbers that represents numerical data on a given date.
+// The first three elements are (year, month, date) and the last is the value.
+// Months are 1-indexed (the way humans write months) and not 0-indexed like Date.getMonth.
+
+// These functions are provided for getting data out of quads.
+function allSchoolYearStarts(dateRange) {
+  // dateRange: A 2-element array of Moment objects.
+  // returns: An array of integers, each school year in the range.
+  const schoolYearStarts = _.map(dateRange, toSchoolYear);
+  return _.range(schoolYearStarts[0], schoolYearStarts[1] + 1);
+}
+
+
+export function toMoment(quad) {
+  return moment.utc([quad[0], quad[1], quad[2]].join('-'), 'YYYY-M-D');
+}
+
+export function toDate(quad) {
+  return toMoment(quad).toDate();
+}
+
+export function toValue(quad) {
+  return quad[3];
+}
+
+export function toPair(quad){
+  return [toMoment(quad).valueOf(), toValue(quad)];
+}
+
+// The "Star Object" adds an additional data point (4-indexed) to a quad.
+// This allows the rendering of gradeLevelEquivalent data in highcharts tooltip
+export function toStarObject(quad){
+  return {
+    x: toMoment(quad).valueOf(),
+    y: toValue(quad),
+    gradeLevelEquivalent: toGradeLevelEquivalent(quad)
```

Comment:
@kevinrobinson I'm learning something here . . . so when we have an JS file that's a list of `export`'ed functions, the functions can reference each other — like `toStarObject` calling `toGradeLevelEquivalent` — and they don't have to be in any particular order, is that correct?

Reply:
Yeah, the way I think about it is these are just plain `function` like normal JS.  And so code can call functions just like normal.  In JS function declarations are all "hoisted" to the top of the function scope so that you can write them in any order you want.  This is different than if you define anonymous functions and assign them to variables like this: `const foo = function() { return 'wat'; }`

And then separately, you can tag `export` in front of functions to make them publicly available to other code that imports the module.

Reply:
Oh that's super helpful, thanks for explaining @kevinrobinson!